scipy/optimize/_differentialevolution.py

```diff
@@ -551,14 +586,34 @@ def __init__(self, func, bounds, args=(),
         if updating in ['immediate', 'deferred']:
             self._updating = updating
 
+        self.vectorized = vectorized
+
         # want to use parallelisation, but updating is immediate
         if workers != 1 and updating == 'immediate':
             warnings.warn("differential_evolution: the 'workers' keyword has"
                           " overridden updating='immediate' to"
                           " updating='deferred'", UserWarning)
             self._updating = 'deferred'
 
+        if vectorized and workers != 1:
+                warnings.warn("differential_evolution: the 'workers' keyword"
+                              " overrides the 'vectorized' keyword")
+                self.vectorized = vectorized = False
```

Comment:
Do you also need to set `workers=1`? I'm getting errors if I set workers=2 with this option on.

Reply:
```
>>> from scipy.optimize import rosen, differential_evolution
>>> bounds = [(0,2), (0, 2), (0, 2), (0, 2), (0, 2)]
>>> result = differential_evolution(rosen, bounds)

>>> result = differential_evolution(rosen, bounds, updating='deferred',
...                                 workers=2, vectorized=True)
>>> result.x, result.fun
(array([1., 1., 1., 1., 1.]), 1.9216496320061384e-19)
```
```
/Users/andrew/Documents/Andy/programming/scipy/installdir/lib/python3.9/site-packages/scipy/optimize/_differentialevolution.py:587: UserWarning: differential_evolution: the 'workers' keyword overrides the 'vectorized' keyword
  warnings.warn("differential_evolution: the 'workers' keyword"
```

Seems fine to me, I'm getting the correct UserWarning.

Reply:
OK, it's working for me now, too.